multi-educator/src/Dscp/MultiEducator/Web/Educator/Handlers.hs

```diff
@@ -0,0 +1,23 @@
+-- | API handlers specific for multi educator.
+
+module Dscp.MultiEducator.Web.Educator.Handlers
+       ( certificatesApiHandlers
+       ) where
+
+import Loot.Base.HasLens (lensOf)
+
+import Dscp.DB.SQL
+import Dscp.Educator.Web.Educator
+import Dscp.MultiEducator.Launcher.Mode
+import Dscp.MultiEducator.Web.Educator.API
+import Dscp.MultiEducator.Web.Educator.Types
+
+certificatesApiHandlers
+    :: forall m ctx. MultiEducatorWorkMode ctx m
+    => CertificatesApiHandlers m
+certificatesApiHandlers = CertificatesApiEndpoints
+    { cGetCertificate = \(CertificateName eId cId) -> do
+            db <- view (lensOf @SQL)
+            setSchemaName db $ educatorSchemaName eId
+            invoke $ educatorGetCertificate cId
```

Comment:
This should be rewritten; it's nonsensical to reconfigure the whole connection pool for each query - only one connection per query should be reconfigured.

Reply:
It seems to be even not thread-safe, isn't it?

Reply:
Haha, yes: `forEachConnection` is not thread-safe, two threads can simultaneously run it and perform different actions for two distinct subsets of connections.

Reply:
Or even cause a deadlock :grimacing:
(since `forEachConnection` grabs all connections before applying a passed function and thus should be used only from one thread - I completely forgot that it's used inside of `setSchemaName`).